src/users/users.controller.spec.ts

```diff
@@ -785,4 +783,290 @@ describe('UsersAdminController', () => {
+      expect(usersService.findById).toHaveBeenCalledWith(userId);
+    });
+
+    it('should throw NotFoundException when user does not exist', async () => {
+      jest.spyOn(usersService, 'paraseEncryptedParams').mockReturnValue({
+        userId,
+        email,
+      });
+
+      jest
+        .spyOn(usersService, 'findById')
+        .mockRejectedValue(
+          new NotFoundException(`User with ID ${userId} not found`),
+        );
+
+      await expect(controller.register(encryptedData)).rejects.toThrow(
+        new NotFoundException('Invalid link'),
+      );
+    });
+
+    it('should throw NotFoundException when decryption fails', async () => {
+      jest
+        .spyOn(usersService, 'paraseEncryptedParams')
+        .mockImplementation(() => {
+          throw new Error('Decryption failed');
+        });
+
+      await expect(controller.register(encryptedData)).rejects.toThrow(
+        new NotFoundException('Invalid link'),
+      );
+    });
+  });
+
+  describe('User Invite Link', () => {
+    const encryptedData = 'encryptedInviteString';
+    const userId = new Types.ObjectId().toString();
+    const email = 'invite@example.com';
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should process an invite link and redirect to the appropriate page', async () => {
+      const mockUser = createUserMock({
+        _id: new Types.ObjectId(userId),
+        email,
+      });
+
+      jest.spyOn(usersService, 'paraseEncryptedParams').mockReturnValue({
+        userId,
+        email,
+      });
+
+      jest.spyOn(usersService, 'findById').mockResolvedValue(mockUser);
+
+      const result = await controller.register(encryptedData);
+
+      expect(result).toEqual({
+        url: `/leads/createLead?email=${email}`,
+      });
+      expect(usersService.paraseEncryptedParams).toHaveBeenCalledWith(
+        encryptedData,
+      );
+    });
+
+    it('should handle invite link with missing parameters correctly', async () => {
+      jest.spyOn(usersService, 'paraseEncryptedParams').mockReturnValue({
+        userId: '',
+        email,
+      });
+
+      const result = await controller.register(encryptedData);
+
+      expect(result).toEqual({
+        url: `/leads/new-user-form?email=${encodeURIComponent(email)}`,
+      });
+    });
+
+    it('should test the parameter parsing function', () => {
+      // Since this is a specific function used by the endpoint, we should test it
+      // This would typically be in a separate test for the service, but including here for completeness
+
+      // Mock the decrypt function since it's an external dependency
+      const decryptMock = jest
+        .fn()
+        .mockReturnValue('userId=123&email=test@example.com');
+      global.decrypt = decryptMock;
+
+      const encryptedParams = 'encoded%20string';
+
+      // We would need to actually inject the real service to test this
+      // const result = usersService.paraseEncryptedParams(encryptedParams);
+
```

Comment:
@oyedeletemitope why do we have alot of comment codes? can we remove them if not used.